inference/core/workflows/core_steps/utils/dominant_color.py

```diff
@@ -0,0 +1,79 @@
+from abc import ABC, abstractmethod
+from typing import List, Literal, Optional, Type, Union
+
+import numpy as np
+import supervision as sv
+from pydantic import AliasChoices, ConfigDict, Field
+from sklearn.cluster import MiniBatchKMeans
+
+from inference.core.workflows.core_steps.visualizations.utils import str_to_color
+from inference.core.workflows.entities.base import OutputDefinition, WorkflowImageData
+from inference.core.workflows.entities.types import (
+    BATCH_OF_STRING_KIND,
+    StepOutputImageSelector,
+    StepOutputSelector,
+    WorkflowImageSelector,
+    WorkflowParameterSelector,
+)
+from inference.core.workflows.prototypes.block import (
+    BlockResult,
+    WorkflowBlock,
+    WorkflowBlockManifest,
+)
+
+TYPE: str = "DominantColor"
+SHORT_DESCRIPTION: str = "Get the dominant color of an image."
+LONG_DESCRIPTION: str = "Get the dominant color of an image as a list of RGB values."
+
+
+class DominantColorManifest(WorkflowBlockManifest):
+    type: Literal[f"{TYPE}"]
+    model_config = ConfigDict(
+        json_schema_extra={
+            "short_description": SHORT_DESCRIPTION,
+            "long_description": LONG_DESCRIPTION,
+            "license": "Apache-2.0",
+            "block_type": "visualization",
+        }
+    )
+
+    image: Union[WorkflowImageSelector, StepOutputImageSelector] = Field(
+        title="Input Image",
+        description="The input image for this step.",
+        examples=["$inputs.image", "$steps.cropping.crops"],
+        validation_alias=AliasChoices("image", "images"),
+    )
+
+    @classmethod
+    def describe_outputs(cls) -> List[OutputDefinition]:
+        return [
+            OutputDefinition(name="rgb_color", kind=[BATCH_OF_STRING_KIND]),
```

Comment:
question: do we see `rgb_color` as separate phenomenon requiring new kind:
* new kind means that on one hand it will not match to other blocks taking strings as inputs
* but as benefit other blocks / workflows elements may be aware of the the new kind and execute special operations (like UQL could have special transformations)
if u provide more details about intended use-case we can decide

Reply:
Good question. This block will take in an image (Crop, Inference Image, etc.) and it calculates the most dominat color and returns the a list of integers for the RGB value of the colors. as of right now, i'm not entirely sure of another block that only takes an image.